frontend/web/components/release-pipelines/SinglePipelineStageAction.tsx

```diff
@@ -24,42 +25,65 @@ const getSegment = (action: StageActionRequest | undefined) => {
 }
 
 interface SinglePipelineStageActionProps {
+  indexPos?: number
   action?: StageActionRequest
-  onActionChange: (option: { value: string; label: string }) => void
-  onSegmentChange: (option: { value: number; label: string }) => void
+  headerText?: string
+  onRemoveAction?: (id: number) => void
+  onActionChange: (option: { value: string; label: string }, id: number) => void
+  onSegmentChange: (
+    option: { value: number; label: string },
+    id: number,
+  ) => void
   isSegmentsLoading: boolean
   segmentOptions: { label: string; value: number }[] | undefined
 }
 
 const SinglePipelineStageAction = ({
   action,
+  headerText = 'Then',
+  indexPos = 0,
   isSegmentsLoading,
   onActionChange,
+  onRemoveAction,
   onSegmentChange,
   segmentOptions,
 }: SinglePipelineStageActionProps) => {
   return (
     <>
       <Row>
         <div className='flex-1 and-divider__line' />
-        <div className='mx-2'>Then</div>
+        <div className='mx-2'>{headerText}</div>
         <div className='flex-1 and-divider__line' />
       </Row>
-      <FormGroup>
+      <FormGroup className='mt-2'>
         <InputGroup
           title='Flag Action'
           component={
-            <Select
-              menuPortalTarget={document.body}
-              maxMenuHeight={120}
-              value={Utils.toSelectedValue(
-                getActionType(action),
-                FLAG_ACTION_OPTIONS,
-                { label: 'Select an action', value: '' },
+            <div className='d-flex align-items-center'>
```

Comment:
I know it's not part of this scope but we should address this soon I think as it relates to yesterday CI problem (the position is sticky so with scroll it follows the viewport):
<img width="354" height="229" alt="image" src="https://github.com/user-attachments/assets/6875591f-10c7-4ba7-b6b1-389ee84277f8" />